library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java

```diff
@@ -185,6 +208,9 @@ public DownloadManagerBuilder withCallbackThrottleByProgressIncrease() {
     }
 
     public DownloadManager build() {
+        if (!notificationChannelProvider.channelId().equals(notificationCreator.channelId())) {
+            throw new DeveloperError("Please ensure to pass same Notification channelId when withNotification methods are used.");
```

Comment:
I think `DeveloperError` is used for an internal error inside the library, this can be influenced by the client so should be an `IllegalArgumentException` or something similar. 

"Please ensure to pass the same Notification `channelId` when calling `withNotification`." might slightly better wording 😺

Reply:
Makes sense. I was not sure while doing this :)

Reply:
Addressed here https://github.com/novoda/download-manager/pull/329/commits/f26a71ddef1071efb513b202f1e551bc4752338f
Also added the same check to Migrator.

Reply:
After applying @danybony 's comments, this check was not necessary at all.